pkg/chunkenc/unordered_test.go

```diff
@@ -276,12 +305,14 @@ func TestHeadBlockInterop(t *testing.T) {
 	// Ensure we can recover ordered checkpoint into unordered headblock with metadata
 	recovered, err = HeadFromCheckpoint(orderedCheckpointBytes, UnorderedWithMetadataHeadBlockFmt)
 	require.NoError(t, err)
-	require.Equal(t, UnorderedWithMetadataHeadBlockFmt, recovered.Format())
-	require.IsType(t, &unorderedHeadBlock{}, recovered)
-	require.IsType(t, unordered.rt, (recovered.(*unorderedHeadBlock)).rt)
-	require.IsType(t, unordered.size, (recovered.(*unorderedHeadBlock)).size)
-	require.IsType(t, unordered.mint, (recovered.(*unorderedHeadBlock)).mint)
-	require.IsType(t, unordered.maxt, (recovered.(*unorderedHeadBlock)).maxt)
+	require.Equal(t, &unorderedHeadBlock{
+		format: UnorderedWithMetadataHeadBlockFmt,
+		rt:     (recovered.(*unorderedHeadBlock)).rt,
+		lines:  (recovered.(*unorderedHeadBlock)).lines,
+		size:   (recovered.(*unorderedHeadBlock)).size,
+		mint:   (recovered.(*unorderedHeadBlock)).mint,
+		maxt:   (recovered.(*unorderedHeadBlock)).maxt,
```

Comment:
I think we should be using data from `unordered` to build the expected value, right?
Otherwise, we won't catch any bugs since we are mostly comparing the same values.

Reply:
Absolutely. Done. Thank you!